csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -386,6 +429,8 @@ internal void UpdateConnectionObservers()
                         bool last = i == connectors.Count - 1;
 
                         TraceLevels traceLevels = _communicator.TraceLevels;
+                        SetHintFailure(connector);
```

Comment:
We should add the entry directly into the dictionary instead of calling a method.

Reply:
fixed